key-generation/src/main/java/com/quorum/tessera/key/generation/AzureVaultKeyGenerator.java

```diff
@@ -0,0 +1,62 @@
+package com.quorum.tessera.key.generation;
+
+import com.quorum.tessera.config.*;
+import com.quorum.tessera.config.keypairs.AzureVaultKeyPair;
+import com.quorum.tessera.key.vault.KeyVaultService;
+import com.quorum.tessera.nacl.Key;
+import com.quorum.tessera.nacl.KeyPair;
+import com.quorum.tessera.nacl.NaclFacade;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class AzureVaultKeyGenerator implements KeyGenerator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AzureVaultKeyGenerator.class);
+
+    private final NaclFacade nacl;
+    private final KeyVaultService keyVaultService;
+
+    public AzureVaultKeyGenerator(final NaclFacade nacl, KeyVaultService keyVaultService) {
+        this.nacl = nacl;
+        this.keyVaultService = keyVaultService;
+    }
+
+    @Override
+    public AzureVaultKeyPair generate(String filename, ArgonOptions encryptionOptions) {
+        final KeyPair keys = this.nacl.generateNewKeys();
+
+        final StringBuilder publicId = new StringBuilder();
+        final StringBuilder privateId = new StringBuilder();
+
+        if(filename != null) {
+            final Path path = Paths.get(filename);
+            final String keyVaultId = path.getFileName().toString();
+
+            if(!keyVaultId.matches("^[0-9a-zA-Z\\-]*$")) {
+                throw new RuntimeException("Generated key ID for Azure Key Vault can contain only 0-9, a-z, A-Z and - characters");
+            }
+
+            if(keyVaultId != null) {
+                publicId.append(keyVaultId);
+                privateId.append(keyVaultId);
+            }
+        }
+
+        publicId.append("Pub");
+        privateId.append("Key");
+
+        saveKeyInVault(publicId.toString(), keys.getPublicKey());
+        saveKeyInVault(privateId.toString(), keys.getPrivateKey());
+
+        return new AzureVaultKeyPair(publicId.toString(), privateId.toString());
+    }
+
+    private void saveKeyInVault(String id, Key key) {
+        keyVaultService.setSecret(id, key.toString());
+        LOGGER.debug("Key {} saved to vault with id {}", key.toString(), id);
```

Comment:
Please don't let me see toString() on a possibly null object again for the logger param there is no need toString will be invoked if obj is not null. java.util.Objects.toString(o) its a null safe way of doing this

Reply:
Noted, toString is no longer used here